src/main/java/org/opentripplanner/graph_builder/module/osm/naming/SidewalkNamer.java

```diff
@@ -0,0 +1,275 @@
+ * by the above algorithm and the sidewalk name remains the same.
+ */
+public class SidewalkNamer implements EdgeNamer {
+
+  private static final Logger LOG = LoggerFactory.getLogger(SidewalkNamer.class);
+  private static final int MAX_DISTANCE_TO_SIDEWALK = 50;
+  private static final double MIN_PERCENT_IN_BUFFER = .85;
+  private static final int BUFFER_METERS = 25;
+
+  private HashGridSpatialIndex<EdgeOnLevel> streetEdges = new HashGridSpatialIndex<>();
+  private Collection<EdgeOnLevel> unnamedSidewalks = new ArrayList<>();
+
+  @Override
+  public I18NString name(OSMWithTags way) {
+    return way.getAssumedName();
+  }
+
+  @Override
+  public void recordEdge(OSMWithTags way, StreetEdge edge) {
+    if (way.isSidewalk() && way.needsFallbackName() && !way.isExplicitlyUnnamed()) {
+      unnamedSidewalks.add(new EdgeOnLevel(edge, way.getLevels()));
+    }
+    if (way.isNamed() && !way.isLink()) {
+      // we generate two edges for each osm way: one there and one back. since we don't do any routing
+      // in this class we don't need the two directions and keep only one of them.
+      var containsReverse = streetEdges
+        .query(edge.getGeometry().getEnvelopeInternal())
+        .stream()
+        .anyMatch(candidate -> candidate.edge.isReverseOf(edge));
+      if (!containsReverse) {
+        streetEdges.insert(
+          edge.getGeometry().getEnvelopeInternal(),
+          new EdgeOnLevel(edge, way.getLevels())
+        );
+      }
+    }
+  }
+
+  @Override
+  public void postprocess() {
+    ProgressTracker progress = ProgressTracker.track(
+      "Assigning names to sidewalks",
+      500,
+      unnamedSidewalks.size()
+    );
+
+    final AtomicInteger namesApplied = new AtomicInteger(0);
+    unnamedSidewalks
+      .parallelStream()
+      .forEach(sidewalkOnLevel -> {
+        assignNameToSidewalk(sidewalkOnLevel, namesApplied);
+
+        // Keep lambda! A method-ref would cause incorrect class and line number to be logged
+        // noinspection Convert2MethodRef
+        progress.step(m -> LOG.info(m));
+      });
+
+    LOG.info(
+      "Assigned names to {} of {} of sidewalks ({}%)",
+      namesApplied.get(),
+      unnamedSidewalks.size(),
+      DoubleUtils.roundTo2Decimals((double) namesApplied.get() / unnamedSidewalks.size() * 100)
+    );
+
+    LOG.info(progress.completeMessage());
+
+    // set the indices to null so they can be garbage-collected
+    streetEdges = null;
+    unnamedSidewalks = null;
+  }
+
+  /**
+   * The actual worker method that runs the business logic on an individual sidewalk edge.
+   */
+  private void assignNameToSidewalk(EdgeOnLevel sidewalkOnLevel, AtomicInteger namesApplied) {
+    var sidewalk = sidewalkOnLevel.edge;
+    var buffer = preciseBuffer(sidewalk.getGeometry(), BUFFER_METERS);
+    var sidewalkLength = SphericalDistanceLibrary.length(sidewalk.getGeometry());
+
+    var envelope = sidewalk.getGeometry().getEnvelopeInternal();
+    envelope.expandBy(0.000002);
```

Comment:
Looking at this again, presumably the envelope is being expanded to catch any edges that might pass through the buffer. At this point you have already constructed the buffer around the geometry of the sidewalk using a specific number of meters. Rather than separately expanding the envelope of the sidewalk by an approximate amount in degrees, you could just use the envelope of the buffered geometry. This ensures you really grab any relevant edges.

Reply:
You're right, this can be replaced: https://github.com/opentripplanner/OpenTripPlanner/pull/5774/commits/70585decabdfe488947bb1ddf4aa7ef207b155ef